sort.h

```diff
@@ -19,6 +19,14 @@
 #define SORT_CMP(x, y)  ((x) < (y) ? -1 : ((y) < (x) ? 1 : 0))
 #endif
 
+#ifndef SORT_SAFE_CPY
```

Comment:
I think I would like to make this user-controlled, and the default be `#define SORT_SAFE_CPY 0`.

The assumption of this library is that you have a contiguous array C types, so standard memmove and memcpy should be fine.

I think if users are relying on `C++` operator overloading of `=`, that is probably an edge case. We can document this as a feature, but not make it standard. I don't want to lose performance in C++ unless users specify this option.